portal-impl/src/com/liferay/portal/language/LanguageResources.java

```diff
@@ -115,6 +121,12 @@ public static String getMessage(Locale locale, String key) {
 			languageMap = _loadLocale(locale);
 		}
 
+		String overrideValue = _getOverrideValue(key, locale);
```

Comment:
@drewbrokke this should be done before the getting of the above "Map<String, String> languageMap", otherwise in case of overriding exist, the above code is just a waste of time.